mem/test_mprotect.c

```diff
@@ -67,9 +68,42 @@ TEST(test_mprotect, test_mprotect_singlecore)
 	TEST_ASSERT_EQUAL(0, munmap(area, page_size * PAGES));
 }
 
+
+TEST(test_mprotect, pages_in_child_copied)
+{
+	unsigned char *area = mmap(NULL, page_size, PROT_READ | PROT_WRITE, MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+	TEST_ASSERT(area != MAP_FAILED);
+
+	area[0] = 0x42;
+
+	TEST_ASSERT_EQUAL_INT(0, mprotect(area, page_size, PROT_READ));
+
+	pid_t pid = fork();
+	TEST_ASSERT(pid >= 0);
+	if (pid == 0) {
+		/* Wait for modifications in parent. */
+		sleep(1);
+		if (area[0] != 0x42) {
+			exit(1);
+		}
```

Comment:
Perhaps it might be easier to do this test in reverse - modify the memory in child instead. This would be nicer, as it would only require `waitpid` sync point, without `sleep`. But I'm not 100% such approach would test exactly the same thing

Reply:
I'll add separate test case for that